src/Tribe/Promoter/Auth.php

```diff
@@ -58,57 +89,24 @@ public function authorize_with_connector() {
 	 *
 	 * @since 4.9.12
 	 *
-	 * @return string The secret key.
-	 *
 	 * @since 4.9
+	 *
+	 * @return string The secret key.
 	 */
 	public function generate_secret_key() {
-		$key = defined( 'AUTH_KEY' ) ? AUTH_KEY : '';
 
-		if ( empty( $key ) ) {
-			$key = $this->generate_key();
+		$salt = wp_generate_password( 6 );
+
+		if ( defined( 'AUTH_KEY' ) ) {
+			$key = AUTH_KEY;
+		} else {
+			$key = wp_generate_password( 25 );
 		}
 
+		$key = sha1( $salt . get_current_blog_id() . $key . get_bloginfo( 'url' ) );
```

Comment:
Suggestion, instead of adding as a string, append each to an array, filter, and then `json_encode` to `sha1`. Would give someone the power to modify their base values for key.

Reply:
How about filtering the generated key instead so in case someone wants to use a custom pattern they can define a custom hashing mechanism as well.

Reply:
Why does this need to be filtered by people outside of Promoter's purposes? Just curious there